### DISCUSSION TITLE
Accessing parameters using PyMechanical

### DISCUSSION THREAD

**Discussion by BohdanHorban** [ansys/pymechanical#539] [Q&A]
Hi All,

how to get all Ansys Mechanical project properties marked as parameters via PyAPI?

**Comment by dipinknair** [comment]
@koubaa @pmaroneh

**Comment by pmaroneh** [comment]
Hi @BohdanHorban , could you please clarify? Maybe by providing a few screenshots? I'm afraid I don't understand your question. Please note that the [Ansys Developer Forum](https://discuss.ansys.com/) is prefered for these types of question.

**Comment by BohdanHorban** [comment]
Hi @pmaroneh , in the screenshot below you can see that we can mark a variable(property) as a parameter. I wonder if we can use PyMechanical/PyAPI to identify this variable among all others?
![Param](https://github.com/ansys/pymechanical/assets/154509349/1f1efcda-bda1-4c3d-bb18-13e74260193f)

**Comment by pmaroneh** [comment]
Hi @BohdanHorban, you can retrieve parameters and values this way: https://discuss.ansys.com/discussion/2719/retrieve-parameters-and-values?utm_source=community-search&utm_medium=organic-search&utm_term=parameters
These commands are for the project page, but you can send them from Mechanical to the project page using the methods explained here: https://discuss.ansys.com/discussion/comment/2699#Comment_2699?utm_source=community-search&utm_medium=organic-search&utm_term=run+Mechanical+project

**Comment by koubaa** [comment]
I don't know if these parameters have any significance for pymechanical, as it is wrapping standalone mechanical which doesn't have access to the Workbench project page.

**Reply by pmaroneh** [reply]
True. As we can't access the WB project page, I guess the only way would be to loop through all items in the tree, check whether they have an input scalar value, and if so check whether that value is a parameter...
